localstack/aws/handlers/service_plugin.py

```diff
@@ -37,32 +39,31 @@ def require_service(self, _: HandlerChain, context: RequestContext, response: Re
         if not context.service:
             return
 
+        service_name: str = context.service.service_name
+        if not self.service_manager.exists(service_name):
+            raise NotImplementedError
+
+        service_operation: Optional[ServiceOperation] = context.service_operation
         request_router = self.service_request_router
 
-        # verify that we have a route for this request
-        service_operation = context.service_operation
+        # Ensure the Service is loaded and set to ServiceState.RUNNING if not in an erroneous state.
+        service_plugin: Service = self.service_manager.require(service_name)
```

Comment:
had to think a bit about the implications on concurrency of moving this out of the lock context, but i think it's safe because ServiceManager is designed to be thread safe :+1: